kevm-pyk/src/kontrol/solc_to_k.py

```diff
@@ -342,17 +352,86 @@ def test_to_claim_name(t: str) -> str:
     def name_upper(self) -> str:
         return self.name[0:1].upper() + self.name[1:]
 
+    @staticmethod
+    def escaped_chars() -> list[str]:
+        return [Contract.prefix_code, '_', '$']
+
+    @staticmethod
+    def escape_char(char: str) -> str:
+        match char:
+            case Contract.prefix_code:
+                as_ecaped = Contract.prefix_code
+            case '_':
+                as_ecaped = 'Und'
+            case '$':
+                as_ecaped = 'Dlr'
+            case _:
+                as_ecaped = hex(ord(char)).removeprefix('0x')
+        return f'{Contract.prefix_code}{as_ecaped}'
+
+    @staticmethod
+    def unescape_seq(seq: str) -> tuple[str, int]:
+        if seq.startswith(Contract.prefix_code + Contract.prefix_code):
+            return Contract.prefix_code, 1
+        elif seq.startswith('Und'):
+            return '_', 3
+        elif seq.startswith('Dlr'):
+            return '$', 3
+        else:
+            return chr(int(seq, base=16)), 4
+
+    @staticmethod
+    def escaped(name: str, prefix: str) -> str:
+        """
+        escape all the chars that would cause issues once kompiling and add a prefix to mark it as "escaped"
+        """
+        escaped = prefix
+        for char in iter(name):
+            if char in Contract.escaped_chars():
+                escaped += Contract.escape_char(char)
+            else:
+                escaped += char
+        return escaped
```

Comment:
Maybe something like this to have less string concatenations?
```suggestion
        escaped = [Contract.escape_char(char) if char in Contract.escaped_chars() else char for char in name]
        return prefix + ''.join(escaped)
```

Reply:
Looks good! Thanks!